pkg/minikube/registry/drvs/hyperkit/hyperkit.go

```diff
@@ -85,5 +92,70 @@ func status() registry.State {
 	}
 
+	// Split version from v0.YYYYMMDD-HH-xxxxxxx or 0.YYYYMMDD to YYYYMMDD
+	currentVersion := convertVersionToDate(string(out))
+	specificVersion := splitHyperKitVersion(getHyperKitVersion())
+	// If current hyperkit is not newer than minimumVersion, suggest upgrade information
+	isNew, err := isNewerVersion(currentVersion, specificVersion)
+	if err != nil {
+		return registry.State{Installed: true, Healthy: true, Error: fmt.Errorf("hyperkit version check failed:\n%s", err), Doc: docURL}
+	}
+	if !isNew {
+		return registry.State{Installed: true, Healthy: true, Error: fmt.Errorf("the installed hyperkit version (0.%s) is older than the minimum recommended version (%s)", currentVersion, getHyperKitVersion()), Fix: "Run 'brew upgrade hyperkit'", Doc: docURL}
+	}
+
 	return registry.State{Installed: true, Healthy: true}
 }
+
+// isNewerVersion checks whether current hyperkit is newer than specific version
+func isNewerVersion(currentVersion string, specificVersion string) (bool, error) {
+	// Convert hyperkit version to time.Date to compare whether hyperkit version is not old.
+	layout := "20060102"
+	currentVersionDate, err := time.Parse(layout, currentVersion)
+	if err != nil {
+		glog.Warningf("Unable to parse to time.Date: %v", err)
+		return false, err
+	}
+	specificVersionDate, err := time.Parse(layout, specificVersion)
+	if err != nil {
+		glog.Warningf("Unable to parse to time.Date: %v", err)
+		return false, err
+	}
+	// If currentVersionDate is equal to specificVersionDate, no need to upgrade hyperkit
+	if currentVersionDate.Equal(specificVersionDate) {
+		return true, nil
+	}
+	// If currentVersionDate is after specificVersionDate, return true
+	return currentVersionDate.After(specificVersionDate), nil
+}
+
+// convertVersionToDate returns current hyperkit version
+// hyperkit returns version info with two patterns(v0.YYYYMMDD-HH-xxxxxxx or 0.YYYYMMDD)
+// convertVersionToDate splits version to YYYYMMDD format
+func convertVersionToDate(hyperKitVersionOutput string) string {
+	// `hyperkit -v` returns version info with multi line.
+	// Cut off like "hyperkit: v0.20190201-11-gc0dd46" or "hyperkit: 0.20190201"
+	versionLine := strings.Split(hyperKitVersionOutput, "\n")[0]
+	// Cut off like "v0.20190201-11-gc0dd46" or "0.20190201"
+	version := strings.Split(versionLine, ": ")[1]
+	// Format to "YYYYMMDD"
+	return splitHyperKitVersion(version)
+}
+
+// splitHyperKitVersion splits version from v0.YYYYMMDD-HH-xxxxxxx or 0.YYYYMMDD to YYYYMMDD
+func splitHyperKitVersion(version string) string {
+	if strings.Contains(version, ".") {
+		// Cut off like "20190201-11-gc0dd46" or "20190201"
+		version = strings.Split(version, ".")[1]
+	}
+	if len(version) >= 8 {
+		// Format to "20190201"
+		version = version[:8]
+	}
+	return version
+}
+
+// getHyperKitVersion returns the minimum hyperKit version
```

Comment:
`Get` is implied for Go function names. This would normally just be called `hyperkit.hyperKitVersion`. I would rather just see this method removed, since it does not add anything over referring  to `minimumVersion` directly.

Reply:
Sorry, you are right.
I removed this getter func.